src/connectors/clear-refinements/connectClearRefinements.ts

```diff
@@ -7,73 +7,83 @@ import {
 });
 
-/**
- * @typedef {Object} CustomClearRefinementsWidgetOptions
- * @property {string[]} [includedAttributes = []] The attributes to include in the refinements to clear (all by default). Cannot be used with `excludedAttributes`.
- * @property {string[]} [excludedAttributes = ['query']] The attributes to exclude from the refinements to clear. Cannot be used with `includedAttributes`.
- * @property {function(object[]):object[]} [transformItems] Function to transform the items passed to the templates.
- */
-
-/**
- * @typedef {Object} ClearRefinementsRenderingOptions
- * @property {function} refine Triggers the clear of all the currently refined values.
- * @property {boolean} hasRefinements Indicates if search state is refined.
- * @property {function} createURL Creates a url for the next state when refinements are cleared.
- * @property {Object} widgetParams All original `CustomClearRefinementsWidgetOptions` forwarded to the `renderFn`.
- */
-
-/**
- * **ClearRefinements** connector provides the logic to build a custom widget that will give the user
- * the ability to reset the search state.
- *
- * This connector provides a `refine` function to remove the current refined facets.
- *
- * The behaviour of this function can be changed with widget options. If `clearsQuery`
- * is set to `true`, `refine` will also clear the query and `excludedAttributes` can
- * prevent certain attributes from being cleared.
- *
- * @type {Connector}
- * @param {function(ClearRefinementsRenderingOptions, boolean)} renderFn Rendering function for the custom **ClearRefinements** widget.
- * @param {function} unmountFn Unmount function called when the widget is disposed.
- * @return {function(CustomClearRefinementsWidgetOptions)} Re-usable widget factory for a custom **ClearRefinements** widget.
- * @example
- * // custom `renderFn` to render the custom ClearRefinements widget
- * function renderFn(ClearRefinementsRenderingOptions, isFirstRendering) {
- *   var containerNode = ClearRefinementsRenderingOptions.widgetParams.containerNode;
- *   if (isFirstRendering) {
- *     var markup = $('<button id="custom-clear-all">Clear All</button>');
- *     containerNode.append(markup);
- *
- *     markup.on('click', function(event) {
- *       event.preventDefault();
- *       ClearRefinementsRenderingOptions.refine();
- *     })
- *   }
- *
- *   var clearRefinementsCTA = containerNode.find('#custom-clear-all');
- *   clearRefinementsCTA.attr('disabled', !ClearRefinementsRenderingOptions.hasRefinements)
- * };
- *
- * // connect `renderFn` to ClearRefinements logic
- * var customClearRefinementsWidget = instantsearch.connectors.connectClearRefinements(renderFn);
- *
- * // mount widget on the page
- * search.addWidgets([
- *   customClearRefinementsWidget({
- *     containerNode: $('#custom-clear-all-container'),
- *   })
- * ]);
- */
-export default function connectClearRefinements(renderFn, unmountFn = noop) {
+export type ClearRefinementsConnectorParamsItem = {
```

Comment:
I asked on slack, but I am quite sure that an item on clear refinements is a simple string.

Reply:
On the test for example, there are simple strings: https://github.com/algolia/instantsearch.js/pull/4385/files#diff-93bf43d5d7e3988ea2803945111d3d80L558.

Reply:
And same on docs: https://www.algolia.com/doc/api-reference/widgets/clear-refinements/js/#widget-param-transformitems.

Reply:
You're right, they're strings. In these cases, you can [try it in a sandbox](https://codesandbox.io/s/instantsearchjs-clear-refinements-qebvy?file=/src/app.js).